### PR TITLE
Adding pre oc 9.0 CardDAV endpoint for migration of old clients

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs #21812

### CODE REVIEW

File: apps/dav/lib/connector/sabre/auth.php [hunk 1]
```diff
@@ -51,10 +51,11 @@ class Auth extends AbstractBasic {
 	 * @param IUserSession $userSession
```
Comment (LukasReschke) [inline apps/dav/lib/connector/sabre/auth.php]: PHPDoc is lying here :speak_no_evil:

File: apps/dav/lib/connector/sabre/principal.php [hunk 2]
```diff
@@ -46,12 +46,18 @@ class Principal implements BackendInterface {
...
+	/** @var string */
+	private $principalPrefix;
+
 	/**
 	 * @param IUserManager $userManager
 	 */
-	public function __construct(IUserManager $userManager, IGroupManager $groupManager) {
+	public function __construct(IUserManager $userManager,
+								IGroupManager $groupManager,
+								$principalPrefix = 'principals/users/') {
```
Comment (LukasReschke) [inline apps/dav/lib/connector/sabre/principal.php]: PHPDoc is lying here as well :speak_no_evil:

File: apps/dav/appinfo/info.xml [hunk 3]
```diff
@@ -15,6 +15,12 @@
 		<files>appinfo/v1/webdav.php</files>
 		<webdav>appinfo/v1/webdav.php</webdav>
 		<dav>appinfo/v2/remote.php</dav>
+		<!-- carddav endpoints as used before ownCloud 9.0 -->
+		<contacts>appinfo/v1/carddav.php</contacts>
+		<carddav>appinfo/v1/carddav.php</carddav>
+		<!-- caldav endpoints as used before ownCloud 9.0 -->
+		<calendar>appinfo/caldav.php</calendar>
+		<caldav>appinfo/caldav.php</caldav>
```
Comment (LukasReschke) [inline apps/dav/appinfo/info.xml]: Broken. You forgot the /v1/ here.

Comment (LukasReschke) [inline apps/dav/appinfo/info.xml]: {"reqId":"UFR5XQTjt6DMngOzEEk9","remoteAddr":"::1","app":"PHP","message":"require_once(): Failed opening required '\/Users\/lukasreschke\/Documents\/Programming\/master\/apps\/dav\/appinfo\/caldav.php' (include_path='\/Users\/lukasreschke\/Documents\/Programming\/master\/3rdparty\/pear\/console_getopt:\/Users\/lukasreschke\/Documents\/Programming\/master\/3rdparty\/pear\/pear_exception:\/Users\/lukasreschke\/Documents\/Programming\/master\/3rdparty\/pear\/pear-core-minimal\/src:\/Users\/lukasreschke\/Documents\/Programming\/master\/3rdparty\/pear\/archive_tar:\/Users\/lukasreschke\/Documents\/Programming\/master\/3rdparty\/phpseclib\/phpseclib\/phpseclib:\/Users\/lukasreschke\/Documents\/Programming\/master\/lib\/private:\/Users\/lukasreschke\/Documents\/Programming\/master\/config:\/Users\/lukasreschke\/Documents\/Programming\/master\/3rdparty:\/Users\/lukasreschke\/Documents\/Programming\/master\/apps:.::\/Users\/lukasreschke\/Documents\/Programming\/master') at \/Users\/lukasreschke\/Documents\/Programming\/master\/remote.php#137","level":3,"time":"2016-02-05T13:14:11+00:00"}

**Comment by LukasReschke** [comment]
Works for me once https://github.com/owncloud/core/pull/21817/files#r52013795 is fixed. :+1:

**Comment by DeepDiver1975** [comment]
@LukasReschke fixed - thx

**Comment by PVince81** [comment]
Code looks good :+1:

**Pull request merged**